context.go

```diff
@@ -0,0 +1,32 @@
+package amqprpc
+
+import "context"
+
+type ctxKey int
+
+const (
+	queueNameKey ctxKey = iota
+	serverKey
+)
+
+// ContextWithQueueName adds the given queueName to the provided context.
+func ContextWithQueueName(ctx context.Context, queueName string) context.Context {
+	return context.WithValue(ctx, queueNameKey, queueName)
+}
+
+// QueueNameFromContext returns the queue name for the current request.
+func QueueNameFromContext(ctx context.Context) (string, bool) {
+	queueName, ok := ctx.Value(queueNameKey).(string)
+	return queueName, ok
+}
+
+// ContextWithServer adds the *Server to the given context.
+func ContextWithServer(ctx context.Context, server *Server) context.Context {
+	return context.WithValue(ctx, serverKey, server)
+}
```

Comment:
This feels almost like abusing the context, do we really want to pass the whole server pointer around? I saw you mentioned that this is how `net/http` does it so I guess it's considered OK then but would it not be possible to just add the channel if we need to let the consumer grab a notifier for shutting down? Or is this seen as lightweight since it's just the pointer and YOLO?

Reply:
Yes, It feels a bit dirty to pass the whole server. I think I'll ignore stdlib and just add the channel. Do you have a name suggestion? `ShutdownChanFromContext(ctx)`?

Reply:
Let's try it and we'll see how well it works!

`ShutdownChanFromContext` sounds very easy to understand, go for it!